### DISCUSSION TITLE
Type error when doing <Box component={Button}>

### DISCUSSION THREAD

**Discussion by Toshinaki** [mantinedev/mantine#5438] [Q&A]
I created a component with `Button` and want to set different css when `size` is changed.

```jsx
<Box
  mod={{ size: props.size }}
  component={Button}
  {...rest}
  className={clsx(classes.button, props.className)}
>
  <Group gap='xs'>
    ...
  </Group>
</Box>
```

```css
.button {
  &[data-size='xs'] {
    ...
  }
}
```

But it complains `Property 'children' does not exist on type 'IntrinsicAttributes & BoxComponentProps & ComponentProp<(<C = "button">(props: PolymorphicComponentProps<C, ButtonProps>) => ReactElement<...>) & Omit<...> & ThemeExtend<...> & ComponentClasses<...> & { ...; }> & Omit<...> & { ...; }'.ts(2322)`

Is this a bug or am I doing something wrong?

**Comment by rtivital** [comment]
You cannot use one polymorphic component in `component` prop of other polymorphic component. If you need that, use `renderRoot` instead.
